sqlmesh/utils/metaprogramming.py

```diff
@@ -179,25 +213,22 @@ def parse_source(func: t.Callable) -> ast.Module:
 
 
 def _decorator_name(decorator: ast.expr) -> str:
+    node = decorator
     if isinstance(decorator, ast.Call):
-        return decorator.func.id  # type: ignore
-    if isinstance(decorator, ast.Name):
-        return decorator.id
-    return ""
+        node = decorator.func
+    return node.id if isinstance(node, ast.Name) else ""
 
 
-def decorators(func: t.Callable, root_node: t.Optional[ast.Module] = None) -> t.List[str]:
-    """Finds a list of all the decorators of a callable."""
+def decorator_vars(func: t.Callable, root_node: t.Optional[ast.Module] = None) -> t.List[str]:
+    """
+    Returns a list of all the decorators of a callable, as well as names of objects that
+    are referenced in their argument list. These objects may be transitive dependencies
+    that we need to include in the serialized python environments.
+    """
+    names: t.List[str] = []
     root_node = root_node or parse_source(func)
-    decorators = []
-
-    for node in ast.walk(root_node):
-        if isinstance(node, (ast.FunctionDef, ast.ClassDef)):
-            for decorator in node.decorator_list:
-                name = _decorator_name(decorator)
-                if name not in IGNORE_DECORATORS:
-                    decorators.append(name)
-    return unique(decorators)
+    _DecoratorDependencyFinder(names).visit(root_node)
```

Comment:
[Nit] Rather than passing the `names` reference to be mutated inside, I think I bit more obvious control flow would be:
```
finder = _DecoratorDependencyFinder()
finder.visit(roo_node)
return unique(finder.dependencies)
```

Reply:
Ah, yeah that's cleaner, good call.